src/explore-education-statistics-common/src/modules/charts/util/createDataSetCategories.ts

```diff
@@ -283,6 +303,10 @@ export default function createDataSetCategories(
   results: TableDataResult[],
   meta: FullTableMeta,
 ): DataSetCategory[] {
+  // Check if the datasets have ordering applied, old charts won't so we can use this to preserve their ordering.
+  const hasDataSetOrder = axisConfiguration.dataSets.some(
+    dataSet => !!dataSet.order,
+  );
```

Comment:
No need to do this here, we can do it inside of `getSortedDataSetCategoriesRange`